zio/jsonio/encode.go

```diff
@@ -0,0 +1,245 @@
+package jsonio
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/nano"
+	"github.com/brimdata/zed/pkg/terminal/color"
+	"github.com/brimdata/zed/zcode"
+	"github.com/brimdata/zed/zson"
+)
+
+var (
+	boolColor   = []byte("\x1b[1m")
+	fieldColor  = []byte("\x1b[34;1m")
+	nullColor   = []byte("\x1b[2m")
+	numberColor = []byte("\x1b[36m")
+	puncColor   = []byte{} // no color
+	stringColor = []byte("\x1b[32m")
+)
+
+type encoder struct {
+	w           io.Writer
+	buf         bytes.Buffer
```

Comment:
A bufio.Writer would be simpler.

Reply:
I believe you would then have to check the error on every write in that case because it might flush to io.Writer and you might get an error there.

Reply:
Looking at the source code it looks like it does cache the first error encountered so perhaps we can get away with using bufio.Writer.